### ISSUE TITLE
Regression with Stub Default and Unknown Type

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#4962]
**Describe the bug**
```python
class Foo:
    x: int = ...
```

in a stub file starting with 1.1.300 fails to type check in strict mode and produces an error of

```
Type of "x" is "Unknown | int"
```

**Expected behavior**
No errors. This is toy example but real example I hit in [typeshed](https://github.com/python/typeshed/pull/10052#discussion_r1167765606) I can't put default value as it's not value valid in stub file. While ... may not be right as .py default value, it should always be accepted as .pyi default value without introducing an Unknown.

**Comment by erictraut** [comment]
Thanks for reporting this. A recent change uncovered a really old bug that wasn't caught by any of our test cases.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.304, which is now published. It will also be included in a future release of pylance.
